includes/class-sensei-lesson.php

```diff
@@ -2557,23 +2560,35 @@ public function lesson_add_existing_questions() {
 
 				foreach ( $questions as $question_id ) {
 
-					++$question_count;
-
 					$quizzes = get_post_meta( $question_id, '_quiz_id', false );
+
 					if ( ! in_array( $quiz_id, $quizzes ) ) {
+						$number_of_added_questions++;
+						++$question_count;
 						add_post_meta( $question_id, '_quiz_id', $quiz_id, false );
 						$lesson_id = get_post_meta( $quiz_id, '_quiz_lesson', true );
 						update_post_meta( $lesson_id, '_quiz_has_questions', '1' );
+						add_post_meta( $question_id, '_quiz_question_order' . $quiz_id, $quiz_id . '000' . $question_count );
+						$question_type = Sensei()->question->get_question_type( $question_id );
+						$return       .= $this->quiz_panel_question( $question_type, $question_count, $question_id );
 					}
-
-					add_post_meta( $question_id, '_quiz_question_order' . $quiz_id, $quiz_id . '000' . $question_count );
-					$question_type = Sensei()->question->get_question_type( $question_id );
-
-					$return .= $this->quiz_panel_question( $question_type, $question_count, $question_id );
 				}
 			}
 		}
 
+		// Log event: when an existing question is added to a quiz.
+		if ( count( $number_of_added_questions ) > 0 ) {
+
+			$event_properties = array(
+				'question_status'   => $question_data['question_status'],
+				'question_type'     => $question_data['question_type'] ? $question_data['question_type'] : 'all',
+				'question_category' => $question_data['question_category'] ? get_term_by( 'slug', $question_data['question_category'], 'question-category' )->term_id : 'all',
```

Comment:
Just noticed this, but I see that `get_term_by` returns `(WP_Term|array|false)`. To be safe, we should check that the return value is not falsy before trying to access `term_id`.

Reply:
Done! Let me know if you'd prefer a different approach or different default values...